File: 3.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        letters = set()
        left = 0
        longest = 0

        for index, letter in enumerate(s):
            while(letter in letters):
                letters.discard(s[left])
                left += 1
            letters.add(letter)
            longest = max(longest, index - left + 1)

        return longest

File: 3.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        unordered_set<char> letters;
        int left = 0, longest = 0;

        for(int i = 0; i < s.size(); i++) {
            while(letters.find(s[i]) != letters.end()) {
                letters.erase(s[left]);
                left++;
            }
            letters.insert(s[i]);
            longest = max(longest, i - left + 1);
        }

        return longest;
    }
};